crates/autopilot/src/database/onchain_order_events/ethflow_events.rs

```diff
@@ -51,22 +51,21 @@ pub struct EthFlowDataForDb {
 impl OnchainOrderParsing<EthFlowData, EthFlowDataForDb> for EthFlowOnchainOrderParser {
     fn parse_custom_event_data(
         &self,
-        contract_events: &[EthContractEvent<ContractEvent>],
+        contract_events: &[(ContractEvent, Log)],
     ) -> Result<Vec<(EventIndex, OnchainOrderCustomData<EthFlowData>)>> {
         contract_events
             .iter()
-            .filter_map(|EthContractEvent { data, meta }| {
-                let meta = match meta {
-                    Some(meta) => meta,
-                    None => return Some(Err(anyhow!("event without metadata"))),
+            .filter_map(|(data, log)| {
+                let Some(_) = log.block_number else {
+                    return Some(Err(anyhow!("event without metadata")));
```

Comment:
Wouldn't it be simpler? I see that below the following approach is also used.
```suggestion
                if log.block_number.is_none() {
                    return Some(Err(anyhow!("event without metadata")));
```